pretix_mollie/payment.py

```diff
@@ -670,6 +669,44 @@ def payment_form_render(self, request) -> str:
         ctx = {'request': request, 'event': self.event, 'settings': self.settings}
         return template.render(ctx)
 
+    def get_due_date(self, payment):
+        # The due date is a little bit fuzzy, since we can (according to the Mollie API) only transmit YYYY-MM-DD,
+        # but internally they will convert it to UTC.
+        # An order.expiry of 05.07.2023 23:59 (GMT+02) would normally be communicated to Mollie as 2023-07-06. Upon
+        # an API-request, Mollie would however return "expiresAt": "2023-07-08T04:00:00+00:00"
+        # Takin into consideration GMT+2/UTC, this still gives an extra 2 hours - but we can't do anything about it.
+        return (payment.order.expires.date() + timedelta(days=1)).isoformat()
+
+    def update_payment_due(self, payment, retry=True):
+        payment_id = payment.info_data.get('id')
+        body = {
+            'dueDate': self.get_due_date(payment),
+        }
+
+        try:
+            req = requests.patch(
+                'https://api.mollie.com/v2/payments/{}'.format(payment_id),
+                json=body,
+                headers=self.request_headers
+            )
+            req.raise_for_status()
+            payment.info_data = req.json()
```

Comment:
Should we create a log entry for this as well?

Reply:
Sure, can do.